### PR TITLE
Remove `context` and `should` from cinstance_test

### PR DESCRIPTION
**Pull request by josemigallas**
### Remove `shoulda-context` (part 1)

First step is removing usage of `context` and `should` in favor of organizing in classes and using `test`.

#### Affected files

* test/unit/cinstance_test.rb

#### JIRA
[THREESCALE-7907: Remove shoulda-context > THREESCALE-7939: Remove all contexts](https://issues.redhat.com/browse/THREESCALE-7939)

### CODE REVIEW

**Review by akostadinov — APPROVED** [review]
Amazing, thank you. I assume non-shoulda classes have been copied verbatim. I didn't verify that.

File: test/unit/cinstance_test.rb [hunk 1]
```diff
@@ -529,12 +333,12 @@ class SuspendTest < ActiveSupport::TestCase
...
-    plan = FactoryBot.create(:application_plan, :issuer => provider_account.first_service!)
+    plan = FactoryBot.create(:application_plan, issuer: provider_account.first_service!)
 
-    cinstance_one = FactoryBot.create(:cinstance, :plan => plan, :user_key => 'foo')
-    cinstance_two = FactoryBot.build(:cinstance, :plan => plan, :user_key => 'foo')
+    FactoryBot.create(:cinstance, plan: plan, user_key: 'foo')
+    cinstance_two = FactoryBot.build(:cinstance, plan: plan, user_key: 'foo')
 
-    assert !cinstance_two.valid?
+    assert_not cinstance_two.valid?
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply test/unit/cinstance_test.rb]: Sure 👍🏻

File: test/test_helpers/cinstance.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module TestHelpers
+  module Cinstance
+    def create_pending_cinstance
+      FactoryBot.create(:cinstance, plan: FactoryBot.create(:application_plan, approval_required: true))
+    end
```
Comment (hallelujah) [inline test/test_helpers/cinstance.rb:7]: You can use a trait instead here https://github.com/3scale/porta/blob/5331e7de089c85a755817b7a0d47ab75b07ba311/test/factories/cinstance.rb

```ruby
FactoryBot.define do
  factory(:contract) do
    association :user_account, :factory => :pending_buyer_account

    after(:stub) do |cinstance|
      cinstance.created_at = 2.months.ago
      cinstance.state = 'live'
      cinstance.user_key ||= SecureRandom.hex(16)
      cinstance.provider_public_key ||= SecureRandom.hex(32)

      cinstance.stubs(:keys).returns([])
      cinstance.stubs(:access_rules).returns([])
    end

    after(:build) do |contract|
      if contract.plan && contract.user_account && contract.user_account.provider_account.nil?
        contract.plan.provider_account.buyer_accounts << contract.user_account
      end
    end
    
    trait :as_pending do
      transient do
        plan { FactoryBot.create(:application_plan, approval_required: true) }
      end
    end
  end

```

Comment (josemigallas) [reply test/test_helpers/cinstance.rb:7]: Cool 👍🏻

**Review by akostadinov — APPROVED** [review]
Changes look good to me. Leaving to @hallelujah the question of a new factory vs a trait.

**Comment by akostadinov** [comment]
Actually I see one failure.

**Review by akostadinov — APPROVED** [review]
I updated factory to match what @hallelujah requested. If you still prefer the full factory, I have no objections but lets discuss in a separate PR.

**Pull request merged**